features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/ElementCallActivity.kt

```diff
@@ -126,14 +129,23 @@ class ElementCallActivity :
 
     @Composable
     private fun ListenToAndroidEvents(pipState: PictureInPictureState) {
+        val lifecycleState by lifecycle.currentStateFlow.collectAsState()
+
         val pipEventSink by rememberUpdatedState(pipState.eventSink)
-        DisposableEffect(Unit) {
-            val onUserLeaveHintListener = Runnable {
-                pipEventSink(PictureInPictureEvents.EnterPictureInPicture)
+        var onUserLeaveHintListener by remember { mutableStateOf<Runnable?>(null) }
+        DisposableEffect(lifecycleState.isAtLeast(Lifecycle.State.STARTED)) {
```

Comment:
I don't understand the `DisposableEffect(lifecycleState.isAtLeast(Lifecycle.State.STARTED))`, this will still launch the effect with value `false`, and then with value `true`

Reply:
I just realised I had added the wrong check to the `if` above (copy&paste issue), so now it should:

- Launch the side effect with `false`, but do nothing.
- Launch the side effect again with `true`, this time register the listener.
- If we go out of the call and then back again, the listener won't be registered again.
- When the activity is finished the listener will also be removed.

I tried with simpler versions using `LaunchedEffect` and a lifecycle observer, but neither of them removed the listener when the activity was closed. Although maybe as the activity is being closed, it's not needed at all.

Reply:
Actually, I was able to simplify it in https://github.com/element-hq/element-x-android/pull/3402/commits/b791c54a9e2d4285d6489bf03caeb1fd86a5d0c4. It seems like using `lifecycleScope.launch` is enough to make it wait until the UI is ready.